cmd/flow-access-server/main.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2021 Optakt Labs OÜ
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"time"
+
+	"github.com/c2h5oh/datasize"
+	"github.com/dgraph-io/badger/v2"
+	"github.com/rs/zerolog"
+	"github.com/spf13/pflag"
+	"google.golang.org/grpc"
+
+	grpczerolog "github.com/grpc-ecosystem/go-grpc-middleware/providers/zerolog/v2"
+	"github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors/logging"
+	"github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors/tags"
+
+	"github.com/onflow/flow/protobuf/go/flow/access"
+	accessApi "github.com/optakt/flow-dps/api/access"
+	dpsApi "github.com/optakt/flow-dps/api/dps"
+	"github.com/optakt/flow-dps/codec/zbor"
+	"github.com/optakt/flow-dps/invoker"
+	"github.com/optakt/flow-dps/models/dps"
+)
+
+const (
+	success = 0
+	failure = 1
+)
+
+func main() {
+	os.Exit(run())
+}
+
+func run() int {
+
+	// Signal catching for clean shutdown.
+	sig := make(chan os.Signal, 1)
+	signal.Notify(sig, os.Interrupt)
+
+	// Command line parameter initialization.
+	var (
+		flagAPI   string
+		flagCache uint64
+		flagChain string
+		flagIndex string
+		flagLevel string
+		flagPort  uint16
+	)
+
+	pflag.StringVarP(&flagAPI, "api", "a", "127.0.0.1:5005", "host URL for GRPC API endpoint")
+	pflag.Uint64VarP(&flagCache, "cache", "e", uint64(datasize.GB), "maximum cache size for register reads in bytes")
+	pflag.StringVarP(&flagChain, "chain", "c", dps.FlowMainnet.String(), "ID of the chain that this server uses as its data source")
```

Comment:
As with the Rosetta API, this value could be deduced by getting `index.First()`, then `index.Header(first)`, and using the Chain ID from the underlying index.